examples/visual_examples/packing/run_rect_packing.py

```diff
@@ -12,54 +12,69 @@
 import numpy as np
 from PIL import Image, ImageDraw, ImageFont
 
-# 2D dimensions of the items to be packed
-widths  = [5, 6, 4, 3, 2, 4, 3, 1, 2, 1, 7, 3]
-heights = [1, 2, 3, 2, 1, 2, 4, 6, 5, 1, 1, 2]
-
-# Number of different items
-n = len(widths)
-
-# Max dimensions of the overall needed space
-max_width_rect = sum(widths)
-max_height_rect = sum(heights)
-
-# Min dimensions of the overall needed space
-min_width_rect = max(widths)
-min_height_rect = max(heights)
-
-# Bounds on needed space (best vs. worst)
-min_area = sum([widths[i]*heights[i] for i in range(0,n)])
-max_area = max_width_rect*max_height_rect
-
-# Decision variables
-rect_height = intvar(min_height_rect, max_height_rect)
-rect_width = intvar(min_width_rect, max_width_rect)
-rect_area = intvar(min_area, max_area)
-x = intvar(0, max_width_rect, shape = n)
-y = intvar(0, max_height_rect, shape = n)
-
-model = Model(
-    ### Necessary constraints
-    (rect_area == rect_height * rect_width), # Definition of the area of the needed space
-    [np.add(x, widths) <= rect_width], # Every item has to be within the width of the overall area
-    [np.add(y, heights) <= rect_height], # Every item has to be withing the height of the overall area
-    [(x[i] + widths[i] <= x[j]) | (x[j] + widths[j] <= x[i]) | (y[i] + heights[i] <= y[j]) | (y[j] + heights[j] <= y[i]) for i in range(n) for j in range(n) if i != j], # Every item has to be fully above, below or next to every other item
-
-    ### Optional constraints
-    # (rect_width > rect_height),
-    # (rect_width >= 10),
-    # (rect_height >= 10)
-)
-
-# Minimize wrt the overall area
-model.minimize(rect_area)
-
-if model.solve():
-    print(rect_height.value())
-    print(rect_width.value())
-    print(rect_area.value())
-    print(x.value())
-    print(y.value())
+def run():
+    # 2D dimensions of the items to be packed
+    widths  = [5, 6, 4, 3, 2, 4, 3, 1, 2, 1, 7, 3]
+    heights = [1, 2, 3, 2, 1, 2, 4, 6, 5, 1, 1, 2]
+
+    # Number of different items
```

Comment:
compute all the deterministaclly computed ones in 'model_rect_packing', so you don't have to pass them as an argument...

wrap the two arrays in cpm_array perhaps again.